### PR TITLE
Add script to seperate out Linux EFI partition

### PR DESCRIPTION
**Pull request by AdityaGarg8**

### CODE REVIEW

File: docs/tools/efi.sh [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	echo "Windows installation detected. Restoring Windows Bootloader."
+	mkdir /boot/efi/EFI/Boot
+	cp /boot/efi/EFI/Microsoft/Boot/bootmgfw.efi /boot/efi/EFI/Boot/bootx64.efi
+fi
+
+# Update fstab
+
+echo "Updating /etc/fstab."
+OLDUUID=$(lsblk -o mountpoints,uuid | grep -w "/boot/efi" | xargs | cut -d " " -f 2)
+sed -i "s/${OLDUUID}/${NEWUUID}/g" /etc/fstab
```
Comment (Redecorating) [inline docs/tools/efi.sh:73]: Can we exit with an error before mounting partitions and moving files if OLDUUID isn't found in fstab?

File: docs/tools/efi.sh [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+mkdir /tmp/newefi
+MOUNTPOINT=/$(lsblk -o label,mountpoints | grep -w "$1" | cut -d "/" -f 2-)
+NEWUUID=$(lsblk -o uuid,label | grep -w "$1" | xargs | cut -d " " -f 1)
+umount "${MOUNTPOINT}" 2>/dev/null || true
+mount UUID="${NEWUUID}" /tmp/newefi
+
+# Copy files
+
+echo "Copying Bootloader to new EFI partition."
+cp -r /boot/efi/EFI /tmp/newefi
```
Comment (Redecorating) [inline docs/tools/efi.sh:45]: can we check that there's an efi partition at /boot/efi and fail early if there isn't?

File: docs/tools/efi.sh [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+if [[ ($input != y) && ($input != Y) ]]
+then
+	echo "Abort!"
+	exit 1
+fi
+
+echo "Changing the EFI partition."
+# Mount the new partition to /tmp/newefi
+
+mkdir /tmp/newefi
```
Comment (Redecorating) [inline docs/tools/efi.sh:36]: maybe confirm that this isn't already there or there's also `mktemp` for making random temporary directories

File: docs/tools/efi.sh (hunk 1 above)
Comment (AdityaGarg8) [reply docs/tools/efi.sh:79]: Check latest push

File: docs/tools/efi.sh (hunk 2 above)
Comment (AdityaGarg8) [reply docs/tools/efi.sh:52]:
```
if [[ ${OLDUUID} != ${FSTABUUID} ]]
then
	echo "Old EFI partition not found in /etc/fstab. Abort!"
	exit 1
fi
```

You should get this error in this case as well.

File: docs/tools/efi.sh (hunk 3 above)
Comment (Redecorating) [reply docs/tools/efi.sh:36]: mktemp makes files and folders with random names so there won't be the issue of if a user has files in /tmp/newefi those files will be deleted.

There's also the option of checking that /tmp/newefi isn't already there.

Comment (Redecorating) [reply docs/tools/efi.sh:36]: actually does mkdir fail if tge directory is already there? if it does then this wouldn't be an issue

Comment (AdityaGarg8) [reply docs/tools/efi.sh:36]: Yes it fails

Comment (AdityaGarg8) [reply docs/tools/efi.sh:36]: I don't think mktemp is required then

**Review by Redecorating — APPROVED** [review]

**Pull request merged**